test/e2e/pkg/infrastructure.go

```diff
@@ -0,0 +1,62 @@
+package e2e
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"os"
+)
+
+// InfrastructureData contains the relevant information for a set of existing
+// infrastructure that is to be used for running a testnet.
+type InfrastructureData struct {
+
+	// Provider is the name of infrastructure provider backing the testnet.
+	// For example, 'docker' if it is running locally in a docker network or
+	// 'digital-ocean', 'aws', 'google', etc. if it is from a cloud provider.
+	Provider string `json:"provider"`
+
+	// Instances is a map of all of the machine instances on which to run
+	// processes for a testnet.
+	// The key of the map is the name of the instance, which each must correspond
+	// to the names of one of the testnet nodes defined in the testnet manifest.
+	Instances map[string]InstanceData `json:"instances"`
+}
+
+// InstanceData contains the relevant information for a machine instance backing
+// one of the nodes in the testnet.
+type InstanceData struct {
```

Comment:
I find "Instance" kinda confusing, I'd call it "Node" (so, NodeData)... but this is just a matter of feeling, so totally up to you

Reply:
:man_shrugging: I find node to be confusing for the same reason. I think of a node as a part of a distributed system, like a Tendermint nod. Instance, in my head is related to ec2 instance, which is a full machine.

Reply:
OK, leave it. Not a big deal, and no clear winner